From ca3b7e5b53970fbb567b937bb465f21a42de404f Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 30 Nov 2020 20:29:14 +0200
Subject: [PATCH] [SW-20154] ppboot/preboot: make sizes platform specific

PPBOOT and PREBOOT binary sizes can be platform specific.
For example, Goya had a requirement of having a small
PPBoot size, so as to reduce the copy latency in BTL
without jeopardising the pcie link bringup time. Hence
move these defines to platform specific headers.

Change-Id: I6805443c8e537c4214f62c49bc6d2ca97f2fcb5b
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi/common.h b/gaudi/common.h
index 0cf5524..eaa76af 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -9,6 +9,8 @@
 
 #include <preboot.h>
 
+#define PPBOOT_SIZE			0x4000		/* 16KB */
+#define PREBOOT_SIZE			0xC000		/* 48KB */
 #define SRAM_SIZE			0x1400000	/* 20MB */
 
 #define HL_SP_SRAM_BASE			0x7FFBFE0000
diff --git a/goya/common.h b/goya/common.h
index 6a64674..68cca8a 100644
--- a/goya/common.h
+++ b/goya/common.h
@@ -11,6 +11,9 @@
 
 #define SRAM_SIZE			0x32A0000	/* 50.625MB */
 
+#define PPBOOT_SIZE			0x4000		/* 16KB */
+#define PREBOOT_SIZE			0xC000		/* 48KB */
+
 #define HL_FW_BOOT_PTRS_BASE		0x60000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xA0000
 
diff --git a/goya2/common.h b/goya2/common.h
index 36c1f56..322e214 100644
--- a/goya2/common.h
+++ b/goya2/common.h
@@ -10,6 +10,9 @@
 #include <goya2_blocks_linux_driver.h>
 #include <goya2/goya2.h>
 
+#define PPBOOT_SIZE			0x4000		/* 16KB */
+#define PREBOOT_SIZE			0xC000		/* 48KB */
+
 #define SRAM_SIZE			0x8000000	/* 128MB */
 #define HL_SP_SRAM_SIZE			0x10000
 
diff --git a/goya2_arc/common.h b/goya2_arc/common.h
index f037021..58a5ffd 100644
--- a/goya2_arc/common.h
+++ b/goya2_arc/common.h
@@ -10,6 +10,9 @@
 #include <preboot.h>
 #include <goya2_blocks_linux_driver.h>
 
+#define PPBOOT_SIZE		0x4000		/* 16KB */
+#define PREBOOT_SIZE		0xC000		/* 48KB */
+
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
 
diff --git a/goya2_arc_gnu/common.h b/goya2_arc_gnu/common.h
index 447d013..d6cdb59 100644
--- a/goya2_arc_gnu/common.h
+++ b/goya2_arc_gnu/common.h
@@ -11,6 +11,9 @@
 
 #define SRAM_SIZE			0x8000000	/* 128MB */
 
+#define PPBOOT_SIZE			0x4000		/* 16KB */
+#define PREBOOT_SIZE			0xC000		/* 48KB */
+
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
 
diff --git a/include/preboot.h b/include/preboot.h
index db9532d..82a474d 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -29,9 +29,6 @@
 #define wfe()	__asm__ __volatile__ ("wfe\n\t" : : )
 #endif
 
-#define PPBOOT_SIZE		0x4000		/* 16KB */
-#define PREBOOT_SIZE		0xC000		/* 48KB */
-
 #define HL_UART_BAUDRATE	115200
 #define HL_SPI_FLASH_MAX_FREQ	25000000
 
